src/utils.rs

```diff
@@ -131,8 +87,12 @@ mod findshlibs_support {
         TargetSharedLibrary::each(|shlib| {
             let maybe_debug_id = shlib
                 .id()
-                .map(|SharedLibraryId::Uuid(bytes)| DebugId::from_uuid(Uuid::from_bytes(bytes)))
-                .or_else(|| find_build_id_from_binary(shlib.name()));
+                .and_then(|id| {
+                    match id {
+                        SharedLibraryId::Uuid(bytes) => Some(DebugId::from_uuid(Uuid::from_bytes(bytes))),
+                        SharedLibraryId::GnuBuildId(ref id) => DebugId::from_guid_age(&id[..16], 0).ok(),
```

Comment:
This might panic if `id` is shorter than 16 bytes. In practice, this is very unlikely to happen, but there's nothing constraining the build id. I'm approving, but update at your discretion.

Fwiw, `DebugId::from_guid_age` requires a length of 16 or errors (no panic). I wouldn't mind to add a `from_gnu` method that takes care of this -- just let me know.